### PR TITLE
update Lido frontend app

### PR DESCRIPTION
**Pull request by mymphe**
This PR includes changes to the Lido frontend application. Among those,
- controls for EL Rewards Withdrawal Limit
- controls for EL Rewards Vault
- display staking limit state

Additionally, the app was restructured and refactored into a more readable and maintainable state.

image

P.S. Sorry for a single commit change, too many moving parts when setting up a local dev environment.

### CODE REVIEW

File: apps/lido/app/src/components/StakingLimitState.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        </LoadableElement>
+      </ListItem>
+      <ListItem label="Max limit">
+        <LoadableElement value={stakingLimitInfo?.maxStakeLimit}>
+          Ξ{formatEth(stakingLimitInfo?.maxStakeLimit)}
+        </LoadableElement>
+      </ListItem>
+      <ListItem label="Restoration rate">
+        <LoadableElement value={stakingLimitInfo?.maxStakeLimitGrowthBlocks}>
+          {stakingLimitInfo?.maxStakeLimitGrowthBlocks} blocks
```
Comment (TheDZhon) [inline apps/lido/app/src/components/StakingLimitState.js:29]: Maybe it's better to show this one as rate (ETH per block), i.e. `maxStakeLimit / maxStakeLimitGrowthBlocks`

File: apps/lido/app/src/components/state/ElRewardsVault.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+const fieldName = 'vault'
+
+const initialValues = {
+  [fieldName]: '',
+}
+
+const validationSchema = yup.object().shape({
+  vault: yup
+    .string('Vault must be a string.')
+    .test(fieldName, 'Vault  must be a valid address.', (vault) => {
```
Comment (TheDZhon) [inline apps/lido/app/src/components/state/ElRewardsVault.js:25]: extra space

File: apps/lido/app/src/components/state/ElRewardsVault.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    <ListItem label="EL Rewards Vault">
+      <Controls>
+        <LoadableElement value={elRewardsVault}>
+          <IdentityBadge entity={elRewardsVault} />
+        </LoadableElement>
+        <IconButton label="edit" icon={<IconEdit />} onClick={openSidePanel} />
+      </Controls>
+      <SidePanel
+        opened={sidePanelOpen}
+        title="Change Withdrawal Credentials"
```
Comment (TheDZhon) [inline apps/lido/app/src/components/state/ElRewardsVault.js:56]: Not sure that this is correct for the `ELRewardsVault` :thinking:

File: apps/lido/arapp.json [hunk 4]
```diff
@@ -15,6 +30,11 @@
       "id": "MANAGE_WITHDRAWAL_KEY",
       "params": []
     },
+    {
+      "name": "Manage protocol contracts role",
+      "id": "MANAGE_PROTOCOL_CONTRACTS_ROLE",
```
Comment (TheDZhon) [inline apps/lido/arapp.json:35]: Maybe we should remove the obsolete roles?
* `SET_ORACLE`
* `SET_TREASURY`
* `SET_INSURANCE_FUND`

**Review by TheDZhon — CHANGES_REQUESTED** [review]
Everything is great :+1: 
I propose to address some minor concerns and suggestions. :pray:

**Comment by mymphe** [comment]
**Update**
1. Addressed all of the suggestions above.

2. Added tooltips for staking parameters
- Limit set - `Staking limit is the total amount of ether that can be staked in a given block. The limit goes down with each deposit but is passively restored on each block.`
- Max limit - `Hard cap on staking limit, i.e. staking limit will not be restored past this limit.`
- Restoration - `Rate at which the staking limit is passively restored on each block up until the max limit if no deposits are submitted.`
@TheDZhon @ujenjt Please feel free to make suggestions

3. Adjusted main vs secondary panel width ratios to accommodate staking parameters, i.e. Lido state panel is a little narrow, and Beacon stats and Staking status panels are a bit wider, see pic below

4. Script optimization - fetch all Lido state parameters in parallel to decrease initial load time


image

**Review by loga4 — APPROVED** [review]

**Review by TheDZhon — APPROVED** [review]
Awesome :+1: 

Maybe change `no deposits are submitted` to `no user funds submitted`.
There is an ambiguity with the `deposits` term :thinking:

**Pull request merged**